File: backend/app/schemas/scn001_history.py
from __future__ import annotations

from datetime import datetime
from typing import Annotated

from pydantic import BaseModel, ConfigDict, Field

from backend.app.schemas.bridge import OverallResult, Severity, SourceScenario


SafeIdentifier = Annotated[str, Field(min_length=1, max_length=120)]
SafeSummary = Annotated[str, Field(min_length=1, max_length=700)]
SafeLabel = Annotated[str, Field(min_length=1, max_length=120)]
SafeAction = Annotated[str, Field(min_length=1, max_length=400)]


class StrictHistorySchema(BaseModel):
    model_config = ConfigDict(
        extra="forbid",
        str_strip_whitespace=True,
        str_min_length=1,
    )


class BeforeReviewJobHistoryItem(StrictHistorySchema):
    before_review_job_id: SafeIdentifier
    status: SafeLabel
    created_at: datetime
    updated_at: datetime
    overall_result: OverallResult | None = None
    overall_severity: Severity | None = None
    summary: SafeSummary | None = None
    has_bridge_run: bool


class BeforeReviewJobHistoryDetail(BeforeReviewJobHistoryItem):
    pass


class BridgeRunHistoryItem(StrictHistorySchema):
    bridge_run_id: SafeIdentifier
    before_review_job_id: SafeIdentifier | None = None
    scenario_id: SafeLabel
    source_scenario: SourceScenario
    user_visible_summary: SafeSummary
    issue_categories: list[SafeLabel] = Field(default_factory=list)
    risk_tags: list[SafeLabel] = Field(default_factory=list)
    law_refs: list[SafeLabel] = Field(default_factory=list)
    recommended_next_actions: list[SafeAction] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/scn001-history.ts
export type Scn001HistoryOverallResult = 'PASS' | 'WARNING' | 'VIOLATION';

export type Scn001HistorySeverity = 'NONE' | 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export type Scn001HistorySourceScenario =
  | 'before_review'
  | 'preset'
  | 'mock'
  | (string & {});

export type BeforeReviewJobHistoryStatus =
  | 'queued'
  | 'running'
  | 'completed'
  | 'failed'
  | (string & {});

export interface BeforeReviewJobHistoryItem {
  before_review_job_id: string;
  status: BeforeReviewJobHistoryStatus;
  created_at: string;
  updated_at: string;
  overall_result: Scn001HistoryOverallResult | null;
  overall_severity: Scn001HistorySeverity | null;
  summary: string | null;
  has_bridge_run: boolean;
}

export type BeforeReviewJobHistoryDetail = BeforeReviewJobHistoryItem;

export interface BridgeRunHistoryItem {
  bridge_run_id: string;
  before_review_job_id: string | null;
  scenario_id: string;
  source_scenario: Scn001HistorySourceScenario;
  user_visible_summary: string;
  issue_categories: string[];
  risk_tags: string[];
  law_refs: string[];
  recommended_next_actions: string[];
  created_at: string;
  updated_at: string;
}

export type BeforeReviewJobHistoryListResponse = BeforeReviewJobHistoryItem[];

export type BridgeRunHistoryListResponse = BridgeRunHistoryItem[];
